src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -269,9 +288,49 @@ def smooth_step(self, unsmoothed_rv, smoothed_rv, start, stop, **kwargs):
             Time-point of the to-be-smoothed RV.
         stop : float
             Time-point of the already-smoothed RV.
+        intermediate_step :
+            Step-size to be taken by approximate transition methods.
+        """
+        if self.dynamic_model.precon is None:
+            return self._smooth_step_classic(
```

Comment:
I can't come up with a better name. I find that `..._preconditioning` and `..._no_preconditioning` look super similar. `classic` is the way I think about it, but please tell me if you dont like it :)

Reply:
I like `classic`, and I also agree that `_preconditioning` and `no_preconditioning` do look similar. An alternative would be to use no suffix at all for the classic method, and just `_preconditioned` otherwise, though explicit is better than implicit.

Reply:
I dont want people to run into the smooth_step_classic thing when they are actually looking for smooth_step. Therefore the explicity (is that a word?). Does that sound reasonable?

Reply:
Yes sure, I'm good with `_classic` and `_with_preconditioning` :+1: